utils/src/activityLog/Activity.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { randomUUID } from "crypto";
+import { EventEmitter, TreeItemCollapsibleState } from "vscode";
+import * as types from '../../index';
+import { parseError } from "../parseError";
+import { AzExtTreeItem } from "../tree/AzExtTreeItem"
+
+export interface ActivityTreeItemOptions {
+    label: string;
+    contextValuePostfix?: string;
+    collapsibleState?: TreeItemCollapsibleState;
+    children?: (parent: types.AzExtParentTreeItem) => AzExtTreeItem[];
+}
+
+export abstract class ActivityBase implements types.Activity {
+
+    public readonly onStart: typeof this._onStartEmitter.event;
+    public readonly onProgress: typeof this._onProgressEmitter.event;
+    public readonly onSuccess: typeof this._onSuccessEmitter.event;
+    public readonly onError: typeof this._onErrorEmitter.event;
+
+    private readonly _onStartEmitter: EventEmitter<types.OnStartActivityData>;
+    private readonly _onProgressEmitter: EventEmitter<types.OnProgressActivityData>;
+    private readonly _onSuccessEmitter: EventEmitter<types.OnSuccessActivityData>;
+    private readonly _onErrorEmitter: EventEmitter<types.OnErrorActivityData>;
+
+    public running: boolean;
+    public done: boolean;
+    public error?: types.IParsedError;
+    public readonly task: types.ActivityTask;
+    public startedAtMs: number;
```

Comment:
Don't need to track `startedAtMs` anymore

Reply:
Sorry, I'm not really following why we wouldn't need to track this anymore. We could use this for timeouts of activities.

Reply:
The purpose of it here was for sorting the tree; which is now handled by `ActivityTreeItem` (in the RG extension). Consumers could still implement their own `Activity` child class that tracks start time and handles timeout.